### ISSUE TITLE
settle on a docstring format and update all docstrings

### ISSUE THREAD

**Issue by bwheelz36** [bayesian-optimization/BayesianOptimization#427]
As discussed in #425.

Suggest using the numpy format, and integrating [pydocstyle](https://numpydoc.readthedocs.io/en/latest/) into test framework

TODO:
- [x] Update `constraint.py` #440 
- [x] Update `logger.py` #435 
- [x] Update `util.py` #435 
- [x] Update `domain_reduction.py` #441 
- [x] Update `observer.py` #445 
- [x] Update `bayesian_optimization.py`#445
- [x] Update `event.py` #445 
- [x] Update `target_space.py` #440 
- [x] Integrate `pydocstyle` into testing. #445 
- [x] Use `numpydoc`

**Comment by bwheelz36** [comment]
this will probably be helpful too:
https://github.com/dadadel/pyment

**Comment by till-m** [comment]
TODO:
- [ ] Update `constraint.py`
- [ ] Update `logger.py`
- [ ] Update `util.py`
- [ ] Update `domain_reduction.py`
- [ ] Update `observer.py`
- [ ] Update `bayesian_optimization.py`
- [ ] Update `event.py`
- [ ] Update `target_space.py`
- [ ] Integrate `pydocstyle` into testing.

**Comment by till-m** [comment]
I have started doing some work on this. I also ran pylint over the codebase in parallel and started fixing some of code smells it reports. Since this is a rather large undertaking I would suggest proceeding in the following way:

- We fork the main branch to a separate branch in this project called something along the lines of `docstring-overhaul` or similar.
- We successively process the docstrings of each file and after finishing 1-3 files open a pull request to the `docstring-overhaul` branch
- After all files have been processed, we merge the `docstring-overhaul` branch into master.

This will make the entire thing more granular and not result in a 4000 lines PR that needs to be reviewed. Any thoughts or objections?
